internal/controller/feature/features.go

```diff
@@ -85,6 +85,13 @@ const (
 	// they know cert-manager will need access to to speed up issuance.
 	// See https://github.com/cert-manager/cert-manager/blob/master/design/20221205-memory-management.md
 	SecretsFilteredCaching featuregate.Feature = "SecretsFilteredCaching"
+
+	// Owner (responsible for graduating feature through to GA): @inteon
+	// GA: v1.13
+	// DontAllowInsecureCSRUsageDefinition will prevent the webhook from allowing
+	// CertificateRequest's usages to be only defined in the CSR, while leaving
+	// the usages field empty.
+	DontAllowInsecureCSRUsageDefinition featuregate.Feature = "DontAllowInsecureCSRUsageDefinition"
```

Comment:
suggestion: I still kinda feel like this shouldn't be a feature gate and should instead be an opt-in flag `--allow-insecure-csr-usage-defintions`. Having a feature gate which is immediate GA in the next available version doesn't feel right either. I don't think it blocks, but it's something others might want to take a look at!

suggestion (non-blocking): If we're going with the feature gate, how about `DisallowInsecureCSRUsageDefinition`? Doesn't matter though!

Reply:
I agree `DisallowInsecureCSRUsageDefinition` sounds better.
I like feature gates because we can remove them later.
I added an issue that is blocking for the 1.13 release to get consensus on this/ or rename this feature flag (both in the controller and webhook): https://github.com/cert-manager/cert-manager/issues/6203